File: 100/121 - Best Time to Buy and Sell Stock.c
#define max(a, b) (a > b ? a : b)
#define min(a, b) (a < b ? a : b)

int maxProfit(int* prices, int pricesSize) {
    int i, profit = 0, min_price = INT_MAX;
    
    for (i = 0; i < pricesSize; ++i) {
        profit = max(profit, prices[i] - min_price);
        min_price = min(min_price, prices[i]);
    }
    
    return profit;
}

File: csharp/121.best-time-to-buy-and-sell-stock.cs
/*
 * @lc app=leetcode id=121 lang=csharp
 *
 * [121] Best Time to Buy and Sell Stock
 */
public class Solution {
    public int MaxProfit(int[] prices) {
        if (prices.Length == 0) {
            return 0;
        }
        int ret = 0, prevMin = prices[0];
        foreach (int price in prices) {
            prevMin = Math.Min(prevMin, price);
            ret = Math.Max(ret, price - prevMin);
        }
        return ret;
    }
}

